pm: clean usage of PM action callback

gmarull opened the pull request:
- Use consistent callback naming, not referencing "pm_control" as it is no longer used
- Update the networking device macros to use action naming
- Fixed some small issues (see commit list)
- Updated docs

gmarull committed:
drivers: sensor: si7210: make PM callback static

The PM action callback is not used externally, so make it static.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: spi: esp32_spim: use NULL for PM callback

A non-existing definition was being used (probably a misspelled
deprecated macro).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: video: ov2640: remove deprecated PM macro

Use NULL if no PM is available.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: use common PM action callback naming

The PM callback is no longer referenced as "pm_control" but
"pm_action_cb", so reflect this new naming on the callbacks.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
tests: use common PM action callback naming

The PM callback is no longer referenced as "pm_control" but
"pm_action_cb", so reflect this new naming on the callbacks.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
net: use pm_action_cb instead of pm_control_cb

The PM action callback is no longer called pm_control_cb but
pm_action_cb.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
doc: reference: power_management: adjust PM callback docs

Documentation was not up-to-date, still referencing pm_control.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
device: remove PM related docs

Remove documentation that belongs to the PM subsystem (already
documented by relevant APIs).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
samples: use common PM action callback naming

The PM callback is no longer referenced as "pm_control" but
"pm_action_cb", so reflect this new naming on the callbacks.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

teburd reviewed (approved):
It's the time to rename, I like the action_cb naming much better than control myself

keith-zephyr reviewed (changes requested):
Can you also delete the `device_pm_control_nop` definition from include/pm/device.h?

gmarull committed:
pm: device: remove deprecated macro device_pm_control_nop

Developers must always use NULL if no device PM is available.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

henrikbrixandersen reviewed (changes requested):
Do we really want to use `*_pm_action_cb()` over just `*_pm_action()`? I mean, all API methods are also callbacks of some sort, but we would not postfix all of those implementations with `_cb`.

gmarull commented:
> Do we really want to use `*_pm_action_cb()` over just `*_pm_action()`? I mean, all API methods are also callbacks of some sort, but we would not postfix all of those implementations with `_cb`.

fair enough, let me just use `_pm_action()`

gmarull committed:
drivers: sensor: bme280: make PM action callback static

The PM action callback is never accessed externally, so make it static.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: serial: npcx: drop inline attribute for PM action callback

Making function inline doesn't provide any benefit here, and is
inconsistent with all other definitions.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

MaureenHelm commented on drivers/ethernet/eth_mcux.c:
```diff
@@ -1395,7 +1395,7 @@ static void eth_mcux_err_isr(const struct device *dev)
 	ETH_NET_DEVICE_DT_INST_DEFINE(n,					\
 			    eth_init,					\
-			    ETH_MCUX_PM_FUNC,				\
+			    eth_mcux_device_pm_action,		\
```
This change looks wrong. You need to keep `ETH_MCUX_PM_FUNC`, which is conditionally defined as `eth_mcux_device_pm_action` or `NULL`

gmarull replied:
fixed, thanks for spotting this one

MaureenHelm commented on drivers/sensor/bq274xx/bq274xx.c:
```diff
@@ -773,7 +773,7 @@ static const struct sensor_driver_api bq274xx_battery_driver_api = {
 	};                                                                     \
 									       \
-	DEVICE_DT_INST_DEFINE(index, &bq274xx_gauge_init, bq274xx_pm_control,  \
+	DEVICE_DT_INST_DEFINE(index, &bq274xx_gauge_init, bq274xx_pm_action,\
```
Please fix the alignment in the trailing backslash

MaureenHelm commented on drivers/sensor/fdc2x1x/fdc2x1x.c:
```diff
@@ -1027,7 +1027,7 @@ static int fdc2x1x_init(const struct device *dev)
 	DEVICE_DT_INST_DEFINE(n,					   \
 			      fdc2x1x_init,				   \
-			      fdc2x1x_device_pm_ctrl,			   \
+			      fdc2x1x_device_pm_action,		   \
```
Fix alignment

MaureenHelm commented on drivers/sensor/ina219/ina219.c:
```diff
@@ -303,7 +304,7 @@ static const struct sensor_driver_api ina219_api = {
 	DEVICE_DT_INST_DEFINE(n,				\
 			      ina219_init,			\
-			      ina219_pm_ctrl,\
+			      ina219_pm_action,		\
```
Fix alignment

MaureenHelm commented on drivers/sensor/lis2mdl/lis2mdl.c:
```diff
@@ -490,7 +490,7 @@ static int lis2mdl_pm_control(const struct device *dev,
 	DEVICE_DT_INST_DEFINE(inst,					\
 			    lis2mdl_init,				\
-			    lis2mdl_pm_control,				\
+			    lis2mdl_pm_action,			\
```
Fix alignment

MaureenHelm commented on drivers/sensor/sgp40/sgp40.c:
```diff
@@ -256,7 +257,7 @@ static const struct sensor_driver_api sgp40_api = {
 	DEVICE_DT_INST_DEFINE(n,				\
 			      sgp40_init,			\
-			      sgp40_pm_ctrl,\
+			      sgp40_pm_action,		\
```
Fix alignment

MaureenHelm commented on drivers/serial/uart_cc13xx_cc26xx.c:
```diff
@@ -537,7 +537,7 @@ static const struct uart_driver_api uart_cc13xx_cc26xx_driver_api = {
 	DEVICE_DT_INST_DEFINE(n,					     \
 		uart_cc13xx_cc26xx_init_##n,				     \
-		uart_cc13xx_cc26xx_pm_control,				     \
+		uart_cc13xx_cc26xx_pm_action,			     \
```
Fix alignment
